server/src/main/java/org/elasticsearch/index/mapper/MetadataFieldMapper.java

```diff
@@ -45,10 +45,12 @@
     }
 
     @SuppressWarnings("rawtypes")
-    public abstract static class Builder<T extends Builder, Y extends MetadataFieldMapper> extends FieldMapper.Builder<T, Y> {
+    public abstract static class Builder<T extends Builder> extends FieldMapper.Builder<T> {
         public Builder(String name, MappedFieldType fieldType, MappedFieldType defaultFieldType) {
             super(name, fieldType, defaultFieldType);
         }
+
+        public abstract MetadataFieldMapper build(BuilderContext context);
```

Comment:
just out of curiosity: this seems to be one of the few reasons where the generic `build` return type was needed?

Reply:
There are a few places in tests as well that expect build() to return a specific type, but nothing within the server code itself.